src/ceph-volume/ceph_volume/util/disk.py

```diff
@@ -364,30 +364,18 @@ def is_device(dev):
         return TYPE in ['disk', 'mpath']
 
     # fallback to stat
-    return _stat_is_device(os.lstat(dev).st_mode)
+    return _stat_is_device(os.lstat(dev).st_mode) and not is_partition(dev)
 
 
-def is_partition(dev):
+def is_partition(dev) -> bool:
```

Comment:
if you introduce python type annotation for this function, you might want to add it for the function signature as well
```suggestion
def is_partition(dev: str) -> bool:
```

Reply:
Thx for the suggestion! Change applied.